pkg/cmd/kafka/topic/consume/consume.go

```diff
@@ -0,0 +1,215 @@
+package consume
+
+import (
+	"errors"
+	"time"
+
+	kafkaflagutil "github.com/redhat-developer/app-services-cli/pkg/cmd/kafka/flagutil"
+
+	"strings"
+
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/kafka/kafkacmdutil"
+	"github.com/redhat-developer/app-services-cli/pkg/core/cmdutil/flagutil"
+	"github.com/redhat-developer/app-services-cli/pkg/core/ioutil/dump"
+	"github.com/redhat-developer/app-services-cli/pkg/core/localize"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/contextutil"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	kafkainstanceclient "github.com/redhat-developer/app-services-sdk-go/kafkainstance/apiv1internal/client"
+	"github.com/spf13/cobra"
+)
+
+const (
+	DefaultOffset    = 0
+	DefaultLimit     = 20
+	DefaultTimestamp = ""
+)
+
+var outputFormatTypes = []string{dump.JSONFormat, dump.YAMLFormat}
+
+type options struct {
+	topicName    string
+	kafkaID      string
+	partition    int32
+	from         string
+	limit        int32
+	offset       int64
+	wait         bool
+	outputFormat string
+
+	f *factory.Factory
+}
+
+// row is the details of a record produced needed to print to a table
+type kafkaRow struct {
```

Comment:
This seems not used?

Reply:
The kafkaRow is used in the mapRecords to row function. [Here](https://github.com/redhat-developer/app-services-cli/blob/produce-consume/pkg/cmd/kafka/topic/consume/consume.go#L203-L209)

Reply:
Sorry for not being clear. What is the purpose of that function. Wy we mapping response to another object that looks very similar to api object?